src/commands/reports/revokeallname.ts

```diff
@@ -0,0 +1,191 @@
+// 		})
+// 		if (!profile || !profile.reports[0])
+// 			return message.reply(
+// 				`Player \`${playername}\` has no profile in your community`
+// 			)
+
+// 		let embed = new MessageEmbed()
+// 			.setTitle("FAGC Profile Revocation")
+// 			.setColor("GREEN")
+// 			.setTimestamp()
+// 			.setAuthor("FAGC Community")
+// 			.setDescription(
+// 				`FAGC Profile of player \`${playername}\` in community ${config.communityId}`
+// 			)
+// 		const fields = await Promise.all(
+// 			profile.reports.map(async (report) => {
+// 				const admin = await this.client.users.fetch(report.adminId)
+// 				return {
+// 					name: report.id,
+// 					value:
+// 						`By: <@${admin.id}> | ${admin.tag}\nBroken rule: ${report.brokenRule}\n` +
+// 						`Proof: ${report.proof}\nDescription: ${report.description}\n` +
+// 						`Automated: ${report.automated
+// 						}\nViolated time: <t:${Math.round(report.reportedTime/1000)}>`,
+// 					inline: true,
+// 				}
+// 			})
+// 		)
+// 		createPagedEmbed(fields, embed, message, { maxPageCount: 5 })
+
+// 		const confirm = await getConfirmationMessage(
+// 			message,
+// 			"Are you sure you want to revoke this player's profile?"
+// 		)
+// 		if (!confirm)
+// 			return message.channel.send("Profile revocation cancelled")
+
+// 		try {
+// 			const response = await this.client.fagc.reports.revokeAllName({
+// 				playername: playername,
+// 				adminId: message.author.id,
+// 				reqConfig: {
+// 					apikey: config.apikey,
+// 				}
+// 			})
+// 			if (response.length) {
+// 				return message.channel.send("Profile revoked!")
+// 			} else {
+// 				return handleErrors(message, response)
+// 			}
+// 		} catch (error) {
+// 			if (error instanceof AuthenticationError)
+// 				return message.channel.send("Your API key is set incorrectly")
+// 			console.error({ error })
+// 			return message.channel.send(
+// 				"Error removing profile. Please check logs."
+// 			)
+// 		}
+// 	}
+// }
+// module.exports = RevokeAllname
+
+const RevokeAllName: Command = {
+	name: "revokeallname",
+	description: "Revokes all reports of a player in your community",
+	aliases: [],
+	category: "reports",
+	usage: "[playername]",
+	examples: ["revokeallname", "revokeallname Potato"],
+	requiresRoles: true,
+	requiredPermissions: ["reports"],
+	requiresApikey: true,
+	async run({ client, message, args, guildConfig }) {
+		if (!guildConfig.apiKey)
+			return message.reply(`${client.emotes.warn} No API key set`)
+
+		// get playername
+		const playername =
+			args.shift() ||
+			(await getMessageResponse(
+				message,
+				`${client.emotes.type} Provide a player name to revoke reports of`,
+			).then((r) => r?.content))
+		if (!playername)
+			return message.channel.send(
+				`${client.emotes.warn} No player name was provided`,
+			)
+
+		// get all their reports and display them on an embed
+		const reports = await client.fagc.reports.search({
+			playername: playername,
+			communityId: guildConfig.communityId,
+		})
```

Comment:
Should check at least one report exists before constructing the embed and making api calls.